chain/jsonrpc/src/lib.rs

```diff
@@ -1358,10 +1356,21 @@ impl JsonRpcHandler {
 fn rpc_handler(
     message: web::Json<Message>,
     handler: web::Data<JsonRpcHandler>,
-) -> impl Future<Output = Result<HttpResponse, HttpError>> {
+) -> impl Future<Output = HttpResponse> {
     let response = async move {
-        let message = handler.process(message.0).await?;
-        Ok(HttpResponse::Ok().json(&message))
+        let message = handler.process(message.0).await;
+        match message.clone() {
+            Message::Response(response) => match response.result {
+                Ok(_) => HttpResponse::Ok().json(message),
+                Err(err) => match err.error_struct {
+                    Some(RpcErrorKind::InternalError(_)) | Some(RpcErrorKind::HandlerError(_)) => {
+                        HttpResponse::InternalServerError().json(&Message::error(err))
+                    }
+                    _ => HttpResponse::Ok().json(&Message::error(err)),
+                },
+            },
+            _ => HttpResponse::InternalServerError().finish(),
+        }
```

Comment:
This might work, but do you want me to get rid of async move and return a simple request from this function?

P.S: No, I'm not changing the JSON response. I'm just modifying the status code of the response

Reply:
> do you want me to get rid of async move and return a simple HttpResponse from this function?

I did not ask for that, but if that will work, I would try that. I don't recall the reason for this manual Future boxing (maybe check git blame first to see the reason for this instead of simple `async fn`). If the code has been there for 3+ years, it could be simply due to no async support in Rust back in those days, and so we can change it now.

P.S. I only discovered this PR due to @race-of-sloths ;-)

Reply:
Done. I've simplified `rpc_handler` function and replaced `async move` with a simple `async fn`. Last commit related to `rpc_handler` function header was done 5 years ago by Illia, so as you said Rust was lacking some core async features back then